### PR TITLE
Get rid of the Scheme-in-C module (lepton core os)

### PR DESCRIPTION
**Pull request by vzh**
- Its functions have been rewritten using Scheme FFI.
- The `platform()` function in `(lepton os)` no longer outputs `carbon` as platform on Macs.
- A simple test for `user-cache-dir()` has been added.
- Renamed wrongly named vars in `lepton-windows.m4`.
- A couple of other little fixes.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -5,6 +5,18 @@ This file documents important user-visible changes in Lepton EDA.  For
...
+-----------------------------------------------
+
+### Changes in `liblepton`:
+
+- The module `(lepton core os)` written in C has been removed.
+  Its functions have been reimplemented using Scheme FFI.
+
+- The Scheme function `platform()` no longer outputs `'(carbon)`.
+  There is no reasonable replacement yet, so the function outputs
+  `'(unknown)` on such machines.
```
Comment (graahnul-grom) [inline NEWS.md:18]: on Mac machines?

Comment (vzh) [reply NEWS.md:18]: Of course, thank you :-)

**Comment by graahnul-grom** [comment]
```
scheme@(guile-user)> %host-type 
$1 = "amd64-portbld-freebsd11.4"
```

Shouldn't we add "freebsd" to the list of platform names? :-)

File: NEWS.md (hunk 1 above)
Comment (vzh) [reply NEWS.md:18]: would be the text intelligible: "on those Mac machines on which it usually did so" ?

**Comment by vzh** [comment]
> ```
> scheme@(guile-user)> %host-type 
> $1 = "amd64-portbld-freebsd11.4"
> ```
> 
> Shouldn't we add "freebsd" to the list of platform names? :-)

I first thought to output "apple" on Macs.  Looking at automake's `config.guess` I believe we better deprecate the `platform()` func.

**Comment by graahnul-grom** [comment]
`./build-tools/config.guess gives` me "amd64-unknown-freebsd11.4"

**Comment by graahnul-grom** [comment]
> I believe we better deprecate the platform() func.

I agree, in scheme scripts we could use that same `%host-type`.
It's not worth trying to duplicate that functionality in our code.

**Comment by graahnul-grom** [comment]
Why `sys-config-dirs` returns the same values as
`sys-data-dirs`? Could you please check it on your system?

**Comment by vzh** [comment]
Because it's the bug in my code :-)  I haven't removed the local variable `pointer` in `c-string-array->list()`.  Thank you!

**Comment by vzh** [comment]
@graahnul-grom, please check it once again.  Hopefully, it'll work correctly now.

**Comment by graahnul-grom** [comment]
Yes, it does. :+1:

**Review by graahnul-grom — APPROVED** [review]

**Comment by vzh** [comment]
OK, thanks!

**Pull request merged**